Update README.md (#19)
Chew/DIRCord: README.md

> ## Setup guide.

Reviewer: ```suggestion
## Setup Guide
```

> 

Reviewer: Put a ### Requirements here, indicate the requirements being Ruby (>= 2.4) and Git

4) Invite the bot to the server, if you don't know how to do that, go [here](https://discordapi.com/permissions.html)  Click on the **Administrator** permission box, and scroll down.  Paste in your *Client ID* that you got from the Discord Developer Portal, and open the link at the bottom of the page.  **DO NOT CHECK ANY OTHER BOXES OTHER OR FILL IN ANYTHING ELSE OTHER THAN THE CLIENT ID.**

5) In the invite link, select your server that you made in Step 1, then invite the bot.

## Here is where the fun begins: *These steps should be universal, but might not work on all devices, these instructions were made for linux, specifically Ubuntu.*

6) Open up your CLI, and run `sudo apt update` to make sure that all of your packages are up to date.  

7) Navigate to your home folder. *(`cd ~`)*

> 8) Make sure you have *git* installed *(`sudo apt install git`)*  Then, `git clone https://github.com/Chew/DIRCord.git` 

Reviewer: Remove the first bit and just keep `git clone` part. Git was already stated as a requirement before.

> 9) CD to the directory, `cd DIRCord/`

Reviewer: ```suggestion
9) Enter the directory with `cd DIRCord`
```

> 10) Make sure you have *ruby* [installed](https://www.ruby-lang.org/en/) and *RVM* [installed](https://rvm.io/)

Reviewer: Remove this, it's stated we need it in the requirements.

> 11) Run `ruby requiregems.rb`  if asked, if you would like to install the gems, respond with *y* for both

Reviewer: ```suggestion
11) Run `ruby requiregems.rb` if prompted to install the gems, respond with *y* for both
```

> 

Reviewer: Put a step here saying to `cp config.example.yaml config.yaml`

> 12) Using your favorite text editor (Such as Atom) or a command line editor, (Such as nano, emacs, or vim) make a new file called *config.yaml* (*For this tutorial, we will use nano*)

Reviewer: ```suggestion
12) Using a text editor (such as Atom) or a command line editor (such as nano, emacs, or vim) open the *config.yaml* file and fill it out
```

> 13) `nano config.yaml` Then either type or paste in the following information:

Reviewer: Remove this step

```yaml
---
nickname: [irc nickname]
server: [irc server]
realname: [(your name) via DIRCord]
username: DIRCord
nickservpass: [nickserv username] [nickserv password]
ssl: 'true'
port: 6697
token: [discord bot's token from Step 3]
server_id: [id of discord server you made in Step 1]
user_id: [your discord user id from Step 2]
```
* nickserv username and password should be the same as your regular IRC account's username and password, so that way you get logged into the same stuff.
* ssl can be set to false, if it is, change the port to *6667* however it might be different depending on your server, contact your server's administrators for help.
**Replace all text in the brackets *[ ]* Including the brackets!**

> 14) Once you're done, press *CTRL + x* to activate nano's shortcuts, then press *y* followed by *enter*.  (This saves the file) 

Reviewer: Remove this step

> 

Reviewer: Add step on how to run

## Setup guide.

1) Create a server on Discord, copy the *server's ID*, you'll need it for later.

2) Copy your *user ID*, you'll need it for later.

> 3) Go to the [Discord Developer Portal](https://discordapp.com/developers/applications/) Make a new application, give it a name, and an avatar if you want.  Copy the *Client ID*, you'll need it for later.  Then head to the **Bot** Section.  Create a bot, scroll down and select the **Administrator** Permission.  Make sure you unselect the slider that makes it a public bot, you don't want anyone but you to have this bot.  Hit the **Copy** button underneath where it says *Bot token*, you'll need this for later.  **DO NOT UNDER ANY CIRCUMSTANCES GIVE *ANYONE* YOUR BOT TOKEN! THIS ALLOWS ANYONE TO TAKE CONTROL OF *YOUR* BOT!**  If you do accidently leak your bot token, do not fear, all hope is not lost.  Simply hit the *Regenerate* button and copy your new token.  **Keep in mind that in order to give it the *Administrator* permission, you *MUST* have 2FA enabled on your discord account.  If you don't know how to do this, go [here](https://support.discordapp.com/hc/en-us/articles/219576828-Setting-up-Two-Factor-Authentication).**

Reviewer: **Keep in mind that in order to give it the *Administrator* permission, you *MUST* have 2FA enabled on your discord account.
what?? since when?? Highly doubt you need this

Reviewer: **DO NOT UNDER ANY CIRCUMSTANCES GIVE *ANYONE* YOUR BOT TOKEN! THIS ALLOWS ANYONE TO TAKE CONTROL OF *YOUR* BOT!** 
Unnecessary.

Reviewer: Make a new application, give it a name, and an avatar if you want.  
Should be
Make a new application, give it a name and an avatar if you want.

4) Invite the bot to the server, if you don't know how to do that, go [here](https://discordapi.com/permissions.html)  Click on the **Administrator** permission box, and scroll down.  Paste in your *Client ID* that you got from the Discord Developer Portal, and open the link at the bottom of the page.  **DO NOT CHECK ANY OTHER BOXES OTHER OR FILL IN ANYTHING ELSE OTHER THAN THE CLIENT ID.**

5) In the invite link, select your server that you made in Step 1, then invite the bot.

## Here is where the fun begins: *These steps should be universal, but might not work on all devices, these instructions were made for linux, specifically Ubuntu.*

6) Open up your CLI, and run `sudo apt update` to make sure that all of your packages are up to date.  

> 7) Navigate to your home folder. *(`cd home/[user]`)*

Reviewer: `cd ~` leads to the home.

8) Make sure you have *git* installed *(`sudo apt install git`)*  Then, `git clone https://github.com/Chew/DIRCord.git` 

9) CD to the directory, `cd DIRCord/`

> 10) Make sure you have *ruby* insalled *(`sudo apt install ruby-full`)*

Reviewer: NEVER RUN THIS COMMAND
Instead, tell the user to just install it.

12) Using your favorite text editor (Such as Atom) or a command line editor, (Such as nano, emacs, or vim) make a new file called *config.yaml* (*For this tutorial, we will use nano*)

13) `nano config.yaml` Then either type or paste in the following information:
```yaml
---
nickname: [irc nickname]
server: [irc server]
realname: [(your name) via DIRCord]
username: DIRCord
nickservpass: [nickserv username] [nickserv password]
ssl: 'true'
port: 6697
token: [discord bot's token from Step 3]
server_id: [id of discord server you made in Step 1]
user_id: [your discord user id from Step 2]
```
* nickserv username and password should be the same as your regular IRC account's username and password, so that way you get logged into the same stuff.
* ssl can be set to false, if it is, change the port to *6667* however it might be different depending on your server, contact your server's administrators for help.
**Replace all text in the brackets *[ ]* Including the brackets!**

14) Once you're done, press *CTRL + x* to activate nano's shortcuts, then press *y* followed by *enter*.  (This saves the file) 

> 15) I would suggest using *tmux* do run this, so you don't have to have a console window constantly open.  (*`sudo apt install tmux`*)

Reviewer: I do not recommend tmux. Please do not put this here.

16) For non-tmux, run `ruby main.rb` boom, you're good to go!

> 17) For tmux, run `tmux new -s DIRCord` then run `ruby main.rb` congrats, now it's running!  To exit the shell without killing the bot, press *CTRL + b* to activate tmux's hotkeys, then press *d* to detach from your shell.  To reattach, use `tmux a -t DIRCord`.

Reviewer: Remove this

6) Open up your CLI, and run `sudo apt update` to make sure that all of your packages are up to date.  

7) Navigate to your home folder. *(`cd ~`)*

8) Make sure you have *git* installed *(`sudo apt install git`)*  Then, `git clone https://github.com/Chew/DIRCord.git` 

9) CD to the directory, `cd DIRCord/`

> 10) Make sure you have *ruby* [insalled](https://www.ruby-lang.org/en/)

Reviewer: Link to RVM

14) Once you're done, press *CTRL + x* to activate nano's shortcuts, then press *y* followed by *enter*.  (This saves the file) 

> 6) For non-tmux, run `ruby main.rb` boom, you're good to go!

Reviewer: step 14, step 6, step 18

also remove "non-tmux"

IRC however, oh boy. On Startup, it will put a BUNCH OF STUFF in the console that probably makes no sense to you, I know it makes no sense to me.

Then, every 2 mins or so, it will "ping" the server, and get a "pong" back.

> ## Commands:

Reviewer: Put the command in backticks, e.g. ~users -> `~users`

Then, every 2 mins or so, it will "ping" the server, and get a "pong" back.

## Prefix

> The bot's default prefix is `~`

Reviewer: No need for this to be separate.

IRC however, oh boy. On Startup, it will put a BUNCH OF STUFF in the console that probably makes no sense to you, I know it makes no sense to me.

Then, every 2 mins or so, it will "ping" the server, and get a "pong" back.

> # Prefix

Reviewer: Put `## Commands`
